substrate/primitives/core/src/paired_crypto.rs

```diff
@@ -71,6 +75,60 @@ pub mod ecdsa_bls377 {
 	impl super::CryptoType for Pair {
 		type Pair = Pair;
 	}
+
+	#[cfg(feature = "full_crypto")]
+	impl Pair {
+		/// hashes the `message` with the specified `MsgHasher` and then signs it using ECDSA
+		/// algorithm. It does not affect the behavoir of BLS12-377 component. It generates
+		/// BLS12-377 Signature according to IETF standard and disregard the hasher for the
+		/// BLS12-377 component
+		pub fn sign_with_hasher<MsgHasher: crate::Hasher>(&self, message: &[u8]) -> Signature
+		where
+			<MsgHasher as crate::Hasher>::Out: Into<[u8; 32]>,
```

Comment:
Why a generic `Hasher`? Isn't this always Keccak?

I ask because in the `Keystore` you will have to expose a function which is not generic over the hasher.
(any `Keystore` implementation should be allowed to be a dyn object, thus generic functions are not supported)

Reply:
So I can't do sign_prehashed because BLS standard doesn't support that. 

It is only `Keccak` for now but 1. [this trait](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/primitives/consensus/beefy/src/lib.rs#L59) requires to support generic hasher. 2. If you want to bridge to bitcoin probably you might want sha256 for example.

We could hard code the keccak in the keystore right? it is better to postpone non-generic-ness as late as possible, no?

Reply:
Yeah sure.